lab/src/Microsoft.ServiceModel.Syndication/src/SyndicationElementExtensionCollection.cs

```diff
@@ -129,21 +129,21 @@ public Collection<TExtension> ReadElementExtensions<TExtension>(string extension
         {
             if (serializer == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("serializer");
+                throw new ArgumentNullException(nameof(serializer));
             }
             return ReadExtensions<TExtension>(extensionName, extensionNamespace, null, serializer);
         }
 
-        internal void WriteTo(XmlWriter writer)
+        internal async Task WriteToAsync(XmlWriterWrapper writer)
         {
```

Comment:
We should only be passing around an XmlWriter as if the writer is already async, there's no need to wrap it.